lib/cli/cli.js

```diff
@@ -5,16 +5,47 @@ const EventEmitter = require('events').EventEmitter;
 const CommandProcessor = require('./command-processor');
 const LineProcessor  = require('./line-processor');
 const utils = require('./utils');
+const serde = require('../serde');
+
+const VERBOSE_LEVEL = {
+  QUIET: 0,
+  NO_HEARTBEAT: 1,
+  ALL: 2,
+};
 
 module.exports = class Cli extends EventEmitter {
   // log - logger function to print results,
   //       help and error messages
-  constructor(log) {
+  constructor(log, verboseLevel) {
     super();
 
     this.log = log;
 
-    this.client = null;
+    if (verboseLevel) {
+      const logger = new EventEmitter();
+
+      logger.on('outgoingMessage', (message) => {
+        if (
+          (!message.pong && !message.ping) ||
+          verboseLevel === VERBOSE_LEVEL.ALL
+        ) {
+          this.log(`Outgoing message:\n${serde.stringify(message)}`);
+        }
+      });
+
+      logger.on('incomingMessage', (message) => {
+        if (
+          (!message.pong && !message.ping) ||
+          verboseLevel === VERBOSE_LEVEL.ALL
+        ) {
+          this.log(`Outgoing message:\n${serde.stringify(message)}`);
+        }
+      });
+
+      this.client = { logger };
+    } else {
+      this.client = null;
```

Comment:
You can avoid this branch completely if you set `this.client` to an empty object before this `if`.